### PR TITLE
feat(onyx-735): open alert bottom sheet when tapping on an alert

### PR DESCRIPTION
**Pull request by nickskalkin**
This PR resolves [ONYX-735]

### Description

Display bottom sheet upon tapping on an alert.

**Not in scope** - artworks counter is a regular text - there is no `<Sup>` component in palette yet. Let's add it in a separate PR. @gkartalis we have pair programming session tomorrow - maybe we can work on this?

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [x] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Display bottom sheet upon tapping on an alert - @nickskalkin 

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-735]: https://artsyproduct.atlassian.net/browse/ONYX-735?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Components/ArtworkLists/views/CreateNewArtworkListView/CreateNewArtworkListView.tsx [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import { useArtworkListsContext } from "app/Components/ArtworkLists/ArtworkListsContext"
 import { ArtworkListsBottomSheetSectionTitle } from "app/Components/ArtworkLists/components/ArtworkListsBottomSheetSectionTitle"
-import { AutoHeightBottomSheet } from "app/Components/ArtworkLists/components/AutoHeightBottomSheet"
 import { useArtworkListsBottomOffset } from "app/Components/ArtworkLists/useArtworkListsBottomOffset"
 import { ArtworkListsViewName } from "app/Components/ArtworkLists/views/constants"
+import { AutoHeightBottomSheet } from "app/Components/BottomSheet/AutoHeightBottomSheet"
```
Comment (nickskalkin) [inline src/app/Components/ArtworkLists/views/CreateNewArtworkListView/CreateNewArtworkListView.tsx:5]: Moved BottomSheet to a common place.

File: src/app/Scenes/SavedSearchAlertsList/Components/AlertBottomSheet.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    useBottomSheetDynamicSnapPoints(initialSnapPoints)
+
+  const navigateToEditScreen = () => {
+    onDismiss()
+    navigate(`settings/alerts/${id}/edit`)
+  }
+
+  const navigateToArtworks = () => {
+    onDismiss()
+    navigate(`settings/alerts/${id}/artworks`)
```
Comment (nickskalkin) [inline src/app/Scenes/SavedSearchAlertsList/Components/AlertBottomSheet.tsx:33]: this screen doesn't exist yet - will add in a separate PR

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Display bottom sheet upon tapping on an alert - @nickskalkin)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 8aba386a8cf44501905ede853e7218b4696acbb8
</p>

**Review by MounirDhahri — APPROVED** [review]
Nice! lgtm

File: src/app/Scenes/SavedSearchAlertsList/Components/SavedSearchesList.tests.tsx [hunk 3]
```diff
@@ -110,4 +111,35 @@ describe("SavedSearches", () => {
...
+                  },
+                },
+              },
+            ],
+          },
+        }),
+      })
+
+      fireEvent.press(screen.getByText("Banksy"))
+      expect(screen.getByText("Edit Alert")).toBeTruthy()
```
Comment (olerichter00) [inline src/app/Scenes/SavedSearchAlertsList/Components/SavedSearchesList.tests.tsx:140]: Personally, I find it easier to read if blocks are separated by an empty line, in this case, the test setup and the assertions.

```suggestion
      fireEvent.press(screen.getByText("Banksy"))

      expect(screen.getByText("Edit Alert")).toBeTruthy()
```

File: src/app/store/config/features.ts [hunk 4]
```diff
@@ -248,6 +248,11 @@ export const features = {
     showInDevMenu: true,
     echoFlagKey: "ARShowBlurhashImagePlaceholder",
   },
+  AREnableAlertBottomSheet: {
+    description: "Enable tapping on alerts to show bottom sheet",
+    readyForRelease: false,
+    showInDevMenu: true,
```
Comment (olerichter00) [inline src/app/store/config/features.ts:254]: Do we need an Echo flag here?

**Review by olerichter00 — APPROVED** [review]
Great work! ⭐

**Review by dariakoko — APPROVED** [review]

File: src/app/store/config/features.ts (hunk 4 above)
Comment (nickskalkin) [reply src/app/store/config/features.ts:254]: I would leave it here until View Artworks screen is implemented. Hopefully I finish it today - then we can get rid of the feature flag

File: src/app/Scenes/SavedSearchAlertsList/Components/SavedSearchesList.tests.tsx (hunk 3 above)
Comment (nickskalkin) [reply src/app/Scenes/SavedSearchAlertsList/Components/SavedSearchesList.tests.tsx:140]: thanks! I will change in a PR that is coming next

**Pull request merged**